### PR TITLE
ROC-7041 RPA callback

### PR DESCRIPTION
**Pull request by akritis22**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-7041

### Change description ###
Allow caseworkers to reset a defendants PIN so that we don't have to do this manually through scripts.
Getting rid of existing RoboticsSupportController

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.cmc.claimstore.services.ccd.Role.CASEWORKER;
+
+@Service
+public class ResetRpaCallbackHandler extends CallbackHandler {
+    private static final String RPA_EVENT_TYPE = "RPAEventType";
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java]: RPAEventTypes should be an enum CaseEvent as an example

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+import uk.gov.hmcts.cmc.domain.models.Claim;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+@Service
+public class RoboticsNotificationServiceImpl implements RoboticsNotificationService {
+
+    private static final String RPA_STATE_MISSING = "missing";
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java]: RPAState should be an enum see CaseEvent as an example

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+    private String resendRPA(
+        String reference,
+        String authorisation,
+        Predicate<Claim> precondition,
+        Consumer<Claim> consumer,
+        String errorMessage
+    ) {
+        String rpaState = RPA_STATE_FAILED;
+        Optional<Claim> claimOptional = claimService.getClaimByReference(reference, authorisation);
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java]:
```suggestion
        Optional<Claim> claimOptional = claimService.getClaimByReference(reference, authorisation)
        .orElseThrow(() -> new BadRequestException(RPA_STATE_MISSING))
```

File: src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/deprectaed/controllers/ResetRpaCallbackHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
+//package uk.gov.hmcts.cmc.claimstore.deprecated.controllers;
```
Comment (andrewwa-kainos) [inline src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/deprectaed/controllers/ResetRpaCallbackHandlerTest.java]: Delete the file please

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java [hunk 5]
```diff
@@ -66,7 +66,8 @@
     CREATE_CITIZEN_CLAIM("CreateCitizenClaim"),
     REVIEWED_PAPER_RESPONSE("ReviewedPaperResponse"),
     MEDIATION_FAILED("FailedMediation"),
-    MEDIATION_SUCCESSFUL("SuccessfulMediation");
+    MEDIATION_SUCCESSFUL("SuccessfulMediation"),
+    RESEND_RPA("ResendRpa");;
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java]: Run checkstyle tasks and action the results, please

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+        if (!claimOptional.isPresent()) {
+            rpaState = RPA_STATE_MISSING;
+            throw new BadRequestException(rpaState);
+        }
+        Claim claim = claimOptional.get();
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java]:
```suggestion
        Claim claim = claimOptional.filter(precondition)
        .orElseThrow(() -> new BadRequestException(RPA_STATE_INVALID));
```

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+    @Autowired
+    public RoboticsNotificationServiceImpl(ClaimService claimService, UserService userService,
+                                           MoreTimeRequestedNotificationService moreTimeRequestedNotificationService,
+                                           DefenceResponseNotificationService defenceResponseNotificationService,
+                                           RequestForJudgementNotificationService ccjNotificationService,
+                                           PaidInFullNotificationService paidInFullNotificationService,
+                                           ResponseDeadlineCalculator responseDeadlineCalculator,
+                                           AppInsightsExceptionLogger appInsightsExceptionLogger,
+                                           DocumentGenerator documentGenerator) {
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java]: _De facto_ standard for method signatures that can't fit on one line is:
* new line after `(`
* each param on its own line
* each param indented a further 4 spaces
* `) {` on its own line aligned with the start of the method

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+        if (!precondition.test(claim)) {
+            rpaState = RPA_STATE_INVALID;
+            throw new BadRequestException(rpaState);
+        }
+        try {
+            consumer.accept(claim);
+            return RPA_STATE_SUCCEEDED;
+        } catch (Exception ex) {
+            appInsightsExceptionLogger.warn(errorMessage, ex);
+            return rpaState + ": " + ex.getMessage();
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java]:
```suggestion
            return RPA_STATE_FAILED + ": " + ex.getMessage();
```
And remove rpaState variable

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

File: src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+import java.util.Map;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ResetRpaCallbackHandlerTest {
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java]: Consider migrating to JUnit 5, if you fancy!

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java (hunk 1 above)
Comment (akritis22) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java]: RPAEventType is a enum, this is just a key used to fetch the object from the map

Comment (daveoh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java]: Sorry, I meant to refer to the constants bellow this line, i.e the ones that are used in the switch statement

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java (hunk 1 above)
Comment (akritis22) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java]: ok got it..changed now

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java (hunk 3 above)
Comment (akritis22) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java]: done

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java (hunk 6 above)
Comment (akritis22) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java]: done

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

File: src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/deprectaed/controllers/ResetRpaCallbackHandlerTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (akritis22) [reply src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/deprectaed/controllers/ResetRpaCallbackHandlerTest.java]: forgot to delete..removed now

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java (hunk 7 above)
Comment (akritis22) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java]: done

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

File: src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+    public void setup() {
+        userService = mock(UserService.class);
+        claimService = mock(ClaimService.class);
+        moreTimeRequestedNotificationService = mock(MoreTimeRequestedNotificationService.class);
+        ccjNotificationService = mock(RequestForJudgementNotificationService.class);
+        responseDeadlineCalculator = mock(ResponseDeadlineCalculator.class);
+        appInsightsExceptionLogger = mock(AppInsightsExceptionLogger.class);
+        documentGenerator = mock(DocumentGenerator.class);
+        defenceResponseNotificationService = mock(DefenceResponseNotificationService.class);
+        paidInFullNotificationService = mock(PaidInFullNotificationService.class);
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java]: If you run with the Mockito JUnit runner (or extend with the mockito extension in JUnit 5) and annotate the fields you don't need to explicitly instantiate them

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaStateType.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaStateType.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaStateType.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaStateType.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaEventType.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/ResetRpaCallbackHandlerTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/rpa/RoboticsNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RpaStateType.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4951336&bid=16365537)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        CaseMapper caseMapper, RoboticsNotificationService roboticsNotificationService) {
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.caseMapper = caseMapper;
+        this.roboticsNotificationService = roboticsNotificationService;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_SUBMIT, this::requestResetRpa
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java]: As discussed. Should be moved to a SUBMIT callback

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+        return response;
+    }
+
+    private Claim convertCallbackToClaim(CallbackRequest callbackRequest) {
+        return caseDetailsConverter.extractClaim(callbackRequest.getCaseDetails());
+    }
+
+    private String handleRoboticsNotification(CallbackRequest callbackRequest, String referenceNumber) {
+        switch (RpaEventType.fromValue(String.valueOf(callbackRequest.getCaseDetails()
+            .getData().get(RPA_EVENT_TYPE)))) {
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ResetRpaCallbackHandler.java]:
```suggestion
        switch (RpaEventType.fromValue((String)callbackRequest.getCaseDetails().getData().get(RPA_EVENT_TYPE))) {
```

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+            "Failed to send CCJ request to RPA");
+    }
+
+    @Override
+    public String rpaPIFNotifications(String referenceNumber) {
+        if (StringUtils.isEmpty(referenceNumber)) {
+            throw new BadRequestException("Reference number not supplied");
+        }
+        User user = userService.authenticateAnonymousCaseWorker();
+        String authorisation = user.getAuthorisation();
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java]: This variable is not used

**Pull request merged**